app/api/retrieveBlogs/route.js

```diff
@@ -1,11 +1,17 @@
-// Pulling mock mongodb
-import mockCollection from "@/data/mockBlogs";
+import { collection, getDocs } from "firebase/firestore";
+
+import { firestore as db } from "@/firebase/firebase";
 
 export async function GET(request) {
-  // TODO: get a connection to MongoDb and return from proper collection
-  const mockCursor = { toArray: () => Promise.resolve(mockCollection) };
-  const mockCollectionObject = { find: () => mockCursor };
-  const documents = await mockCollectionObject.find().toArray();
+  const querySnapshot = await getDocs(collection(db, "blogs"));
+
+  // TODO: we are pulling all blogs for now
+  // TODO: may need to set up different endpoints for filters?
+  const blogs = [];
+  querySnapshot.forEach((doc) => {
+    const data = doc.data();
+    blogs.push(data.id, data.authorName, data.blogName, data.dateCreated);
```

Comment:
maybe change to `createdAt` to be consistent with elsewhere? I realize now I named it `dateCreated` for some reason in `app/api/createBlog/route.js`, so maybe change it there too

Reply:
oh wait, just realizing this is pushing all attributes as separate array elements. instead you'd want something like `blogs.push({data.id, data.authorName, data.blogName, data.dateCreated});` to push each `blog` as an object to the array

Reply:
actually you could just replace this with

```
    const data = doc.data();
    blogs.push(data);
```

since we're using the whole object anyway